extension/bootstrap.js

```diff
@@ -91,25 +169,12 @@ class CopyController {
     if (cmd === "cmd_copy") {
       studyUtils.telemetry({ event: "copy" });
       const shareButton = this.browserWindow.shareButton;
-      if (shareButton !== null && // the button exists
-          shareButton.getAttribute("disabled") !== "true" && // the page we are on can be shared
-          shareButton.getAttribute("cui-areatype") === "toolbar" && // the button is in the toolbar
-          shareButton.getAttribute("overflowedItem") !== "true") { // but not in the overflow menu
-        // check to see if we should call a treatment at all
-        const numberOfTimeShown = Preferences.get("extensions.sharebuttonstudy.counter", 0);
-        if (numberOfTimeShown < MAX_TIMES_TO_SHOW) {
-          Preferences.set("extensions.sharebuttonstudy.counter", numberOfTimeShown + 1);
-
-          if (this.treatment === "ALL") {
-            Object.keys(TREATMENTS).forEach((key, index) => {
-              if (Object.prototype.hasOwnProperty.call(TREATMENTS, key)) {
-                TREATMENTS[key](this.browserWindow, shareButton);
-              }
-            });
-          } else if (this.treatment in TREATMENTS.keys()) {
-            TREATMENTS[this.treatment](this.browserWindow, shareButton);
-          }
-        }
+      // check to see if we should call a treatment at all
+      const numberOfTimeShown = Preferences.get(COUNTER_PREF, 0);
+      if (numberOfTimeShown < MAX_TIMES_TO_SHOW &&
+          Object.keys(TREATMENTS).includes(this.treatment)) {
```

Comment:
This could happen because the treatment pref is set to an invalid treatment or the config file has invalid treatment names, where the valid treatment names are given by the `TREATMENTS` object in `bootstrap.js`. There is no other check to make sure the treatments that are set in the pref or config file are valid.

However, I think that you are right that this check should not occur right before the treatment function is being called and instead be called somewhere more appropriate such as after we choose the variation.